crates/uv-fs/src/path.rs

```diff
@@ -94,55 +94,99 @@ pub fn canonicalize_executable(path: impl AsRef<Path>) -> std::io::Result<PathBu
     }
 }
 
-/// Returns `true` if this is a Python executable installed via the Windows Store, like:
+/// Returns `true` if this is a Python executable or shim installed via the Windows Store, based on
+/// the path.
 ///
-///     `C:\Users\crmar\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.11_qbs5n2kfra8p0\python.exe`.
+/// This method does _not_ introspect the filesystem to determine if the shim is a redirect to the
+/// Windows Store installer. In other words, it assumes that the path represents a Python
+/// executable, not a redirect.
 fn is_windows_store_python(path: &Path) -> bool {
-    if !cfg!(windows) {
-        return false;
-    }
+    /// Returns `true` if this is a Python executable shim installed via the Windows Store, like:
+    ///
+    ///     `C:\Users\crmar\AppData\Local\Microsoft\WindowsApps\python3.exe`
+    fn is_windows_store_python_shim(path: &Path) -> bool {
+        let mut components = path.components().rev();
+
+        // Ex) `python.exe`, or `python3.exe`, or `python3.12.exe`
+        if !components
+            .next()
+            .and_then(|component| component.as_os_str().to_str())
+            .is_some_and(|component| component.starts_with("python"))
+        {
+            return false;
+        }
 
-    if !path.is_absolute() {
-        return false;
-    }
+        // Ex) `WindowsApps`
+        if !components
```

Comment:
This can be simplified to using components.ends_with("Microsoft/WindowsApps") after having done the fuzzy python.exe check